src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -46,6 +43,7 @@
 
 class NatsConnection implements Connection {
 
+    private final static NatsLoggerFacade LOGGER = NatsLoggerFacade.getLogger(NatsConnection.class);

```

Comment:
LOGGER cannot be static. It must be specific to the instance of the connection, because the dev may want to log connections differently.

Reply:
I think howver you meant someting differnt - you meant that every connection could have its own "logger" implementation that differes from others.. If yes then this has to change maybe on another level as well (in the facade) ... Will provide a change for this.